src/ZPublisher/tests/test_WSGIPublisher.py

```diff
@@ -631,6 +636,30 @@ def testHandleErrorsFalseBypassesExceptionResponse(self):
         with self.assertRaises(Unauthorized):
             self._callFUT(environ, start_response, _publish)
 
+    def testDebugExceptionsBypassesExceptionResponse(self):
+        from zExceptions import BadRequest
+
+        # Register an exception view for BadRequest
+        registerExceptionView(IException)
+        environ = self._makeEnviron()
+        start_response = DummyCallable()
+        _publish = DummyCallable()
+        _publish._raise = BadRequest('debugbypass')
+
+        # Responses will always have debug_exceptions set
+        def response_factory(stdout, stderr):
+            response = DummyResponse()
+            response.debug_exceptions = True
+            return response
+
+        # With debug_exceptions, the exception view is not called.
+        with self.assertRaises(BadRequest):
+            self._callFUT(environ, start_response, _publish,
+                          _response_factory=response_factory)
+
+        # Clean up view registration
+        unregisterExceptionView(IException)
```

Comment:
I think there should be a `try-finally` block, so the `unregisterExceptionView` is even called when the tests might fail so it does not mess up other tests.

Reply:
try/finally added